providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java

```diff
@@ -119,27 +118,26 @@ public List<Hook> getProviderHooks() {
     @Override
     public void initialize(EvaluationContext evaluationContext) throws Exception {
         synchronized (eventsLock) {
-            if (eventsLock.initialized) {
+            if (eventsLock.isInitialized()) {
                 return;
             }
 
             flagResolver.init();
+            // block till ready - this works with deadline fine for rpc, but with in_process
+            // we also need to take parsing into the equation
+            // TODO: evaluate where we are losing time, so we can remove this magic number -
+            eventsLock.waitForInitialization(this.deadline * 2);
```

Comment:
We purposely moved this out of the sync block, as the eventhandler also syncs on the event lock. is this now a problem that this is within the sync block?